Physical_Engine/Modify/SetInternalElements2D.cs

```diff
@@ -42,9 +42,15 @@ public static partial class Modify
         [Input("surface", "The ISurface to replace the internal elements of.")]
         [Input("internalElements2D", "New internal elements, will replace any existing ones.")]
         [Output("surface", "The ISurface with new internal elements.")]
-        public static oM.Physical.Elements.ISurface SetInternalElements2D(this oM.Physical.Elements.ISurface surface, List<IElement2D> internalElements2D)
+        public static ISurface SetInternalElements2D(this oM.Physical.Elements.ISurface surface, List<IElement2D> internalElements2D)
         {
-            oM.Physical.Elements.ISurface pp = surface.ShallowClone();
+            if(surface == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Cannot set the internal 2D elements of a null surface.");
+                return surface;
```

Comment:
shouldn't it be safer to return `null`? a general question regarding all code changes resulting in returning a reference to `null` instead of a `null` itself.

Reply:
The surface is null based on the if-check, so `return surface` is the same as `return null`.

Reply:
Again, why not return `null`? `return surface` adds complexity to a simple `return null` IMO. I really do not see why would this method not follow the regular `return null` patter - the more coherent we are, the easier the maintenance will be.

Reply:
Because the Surface is `null` and is typically what gets returned on modify methods. If you dislike it that much you are welcome to do a suggested change and commit it - co-authoring of these PRs is allowed/encouraged as @adecler and I did for Reflection_Engine. I don't mind either way, it's consistent to how it was done on other engines but if you'd prefer `return null` I won't be upset if it's put in :smile: (I'll only be upset if I have to do it and it then holds things up for another day or so as a result :wink: )